tensorflow/lite/micro/kernels/xtensa_hifi/depthwise_conv.cc

```diff
@@ -0,0 +1,597 @@
+      &output_activation_max);
+
+  if((params->dilation_width_factor == 1) && (params->dilation_height_factor == 1)) {
+    const float *input_data, *filter_data, *bias_data;
+    float *output_data;
+    const RuntimeShape& input_shape = GetTensorShape(input);
+    const RuntimeShape& filter_shape = GetTensorShape(filter);
+    const RuntimeShape& output_shape = GetTensorShape(output);
+    const RuntimeShape& bias_shape = GetTensorShape(bias);
+
+    input_data = GetTensorData<float>(input);
+    filter_data = GetTensorData<float>(filter);
+    bias_data = GetTensorData<float>(bias);
+    output_data = GetTensorData<float>(output);
+
+    const int stride_width = params->stride_width;
+    const int stride_height = params->stride_height;
+    const int dilation_width_factor = 1;
+    const int dilation_height_factor = 1;
+    //const int dilation_width_factor = params->dilation_width_factor;;
+    //const int dilation_height_factor = params->dilation_height_factor;
+    const int pad_width = data->padding.width;
+    const int pad_height = data->padding.height;
+    const int depth_multiplier = params->depth_multiplier;
+    TFLITE_DCHECK_EQ(input_shape.DimensionsCount(), 4);
+    TFLITE_DCHECK_EQ(filter_shape.DimensionsCount(), 4);
+    TFLITE_DCHECK_EQ(output_shape.DimensionsCount(), 4);
+
+    const int batches = MatchingDim(input_shape, 0, output_shape, 0);
+    const int output_depth = MatchingDim(filter_shape, 3, output_shape, 3);
+    const int input_height = input_shape.Dims(1);
+    const int input_width = input_shape.Dims(2);
+    const int input_depth = input_shape.Dims(3);
+    const int filter_height = filter_shape.Dims(1);
+    const int filter_width = filter_shape.Dims(2);
+    const int output_height = output_shape.Dims(1);
+    const int output_width = output_shape.Dims(2);
+    const int filter_depth = filter_shape.Dims(3);
+    TFLITE_DCHECK_EQ(output_depth, input_depth * depth_multiplier);
+    TFLITE_DCHECK_EQ(bias_shape.FlatSize(), output_depth);
+
+    int32_t err, input_data_format = 0, output_data_format = 0;
+    void *p_scratch;
+    float *p_filter;
+    int filter_depth_padded, filter_size_padded, required_scratch;
+    int input_precision = PREC_F32;
+    int h, c, i;
+
+    ALLOCATE_XTENSA_NNLIB_SCRATCH_MEM;
+    p_scratch = xtensa_nnlib_scratch_buf;
+
+    filter_depth_padded = (filter_depth + 1) & (~1);
+    filter_size_padded = filter_height * filter_width * filter_depth_padded;
+
+    required_scratch =  xa_nn_conv2d_depthwise_getsize(
+        input_height,
+        input_width,
+        input_depth,
+        filter_height,
+        filter_width,
+        depth_multiplier,
+        stride_width,
+        stride_height,
+        pad_width,
+        pad_height,
+        output_height,
+        output_width,
+        input_precision,
+        input_data_format);
+
+    if(required_scratch <= 0) {
+      context->ReportError(context, "DepthwiseConvFloat: xa_nn_conv2d_depthwise_getsize failed");
```

Comment:
`TF_LITE_KERNEL_LOG should be used instead of context->ReportError, so log strings can be removed to save space if needed.`